strawberry/types/types.py

```diff
@@ -111,6 +114,15 @@ def copy_with(
             (self.origin,),
             {"__strawberry_definition__": new_type_definition},
         )
+
+        # update all annotations to point to the new type
+        for name, type_ in self.origin.__annotations__.items():
+            if is_type_var(type_):
+                actual_type = type_var_map[type_.__name__]
+
+                new_type.__annotations__[name] = actual_type
+                new_type.__dataclass_fields__[name].type = actual_type
```

Comment:
add comments here too, `new_type.__annotations__[name]` is not needed for this use case but it makes sense to be consistent